### PR TITLE
fix: duplicate param_id should be invalid only in profile

### LINKED ISSUE #1339: Duplicate param-ids should be allowed by validation for certain models
## Describe the bug

Currently, `duplicates_validator` does not allow to have duplicate `param_id` at all. However, for certain models such as component-definitions it should be allowed

## To Reproduce

Run `trestle validate --all` for ocp-cd repository.
Observe that validation fails for component-definition.

## Expected behavior

In this case duplicate param_ids should be allowed

## Environment

Latest develop

### PR DESCRIPTION
**Pull request by fsuits**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary
duplicate validator was failing for ssp and compdef if duplicate param_ids were found.  

closes #1339 

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

File: tests/trestle/core/commands/validate_test.py [hunk 1]
```diff
@@ -248,6 +251,15 @@ def test_validate_dup_uuids(sample_component_definition: ComponentDefinition) ->
...
     assert not validator.model_is_valid(sample_component_definition, True)
 
+    assert validator.model_is_valid(sample_trestle_profile, True)
+
+    # add extra set_param with duplicate param_id and confirm it is not valid
+    # only profile has this additional test
+    set_param = sample_trestle_profile.modify.set_parameters[0].copy()
+    set_param.values = ['foo']
+    sample_trestle_profile.modify.set_parameters.append(set_param)
+    assert not validator.model_is_valid(sample_trestle_profile, True)
```
Comment (enikonovad) [inline tests/trestle/core/commands/validate_test.py:261]: May be add the test to assert that validation passes for the `sample_component_definition` from above if that cd has duplicate param_id. Just to solidify this logic for the future.

**Review by enikonovad — APPROVED** [review]
LGTM - one additional test for cd would be nice to ensure this logic lives through refactoring
